### DISCUSSION TITLE
Change the color/highlight the row upon click

### DISCUSSION THREAD

**Discussion by Excelautomation** [filamentphp/filament#9559] [Help]
Using Filament Table, I want to change the color/highlight the specific row, whenever the user clicks on it. This change should happen without reloading the entire table. So basically I just want to add a class to the row instance, and remove it on all other rows.

Using the action method, it seems possible to use ->js(), but I only have access to the record and the livewire object. How can I append this extra class to the row? - I can use some find logic in my javascript expression, but it will be a very nasty solution.

I've been struggling with finding a proper solution for it.

Anyone knows how I can accomplish this?

**Comment by HayderHatem** [comment]
still not getting answer ?

**Comment by rixdev** [comment]
You may have already worked this out, but the solution to this is to use the recordClasses() method on your table. I saw your post during a google and figured it was worth coming back with a solution. 

Something like this in your table component Class or your resource page:


```php
public ?string $selectedRowId = null;

public function table(Table $table): Table
{
    return $table
        ->query($this->getTableQuery())
        ->columns([
            TextColumn::make('example')
                ->sortable()
                ->action(fn (Model $record) => $this->selectRecord($record->id))
                ->size('xs'),
        ])
        ->recordClasses(fn (Model $record) =>
            $record->id === $this->selectedRowId
                ? 'bg-success-500 text-white font-bold border-l-4'
                : ''
        );
}

public function selectRecord(string $id): void
{
    $this->selectedRowId = $id;
    // Dispatch any events you may need here        
}

public function updatedSelectedRowId(): void
{
    $this->render();
}
```

**Reply by Excelautomation** [reply]
Great solution! It's cleaner than the one I came up with.

I ran into a small issue, though. In your solution, $selectedRowId is defined as a string, but my $record->id is an integer. Since you're using ===, the condition is never met.

I had to either change === to == or declare public ?int $selectedRowId = null;
